aquifer-git.js

```diff
@@ -100,185 +98,96 @@ module.exports = (Aquifer, AquiferGitConfig) => {
       return;
     }
 
-    // If we have a name without an email, or an email with no name (like XOR),
-    // then we cannot create a custom signature and need to bail out.
-    if (!options.name !== !options.email) {
-      callback('Both name and email options are required for a custom commit signature.');
-      return;
-    }
-
     // Create the destination directory and initiate the promise chain.
     mktemp.createDir('aquifer-git-XXXXXXX')
+    // Clone the repository.
+    .then((destPath_) => {
+      Aquifer.console.log('Cloning the repository into ' + destPath_ + '...', 'status');
+      destPath = destPath_;
+      return run.invoke('git clone ' + options.remote + ' ' + destPath);
+    })
+
+    // Prepare the repository for the build.
+    .then(() => {
+      Aquifer.console.log('Checking out branch: ' + options.branch + '...', 'status');
+      return run.invoke('bash -c "cd ' + path.join(Aquifer.projectDir, destPath) + ' && git checkout ' + options.branch + '"');
```

Comment:
@chasingmaxwell Just a thought, but it doesn't seem to difficult to add another parameter(s) to `run.invoke()`